apps/bfd-model/bfd-model-rif/src/main/resources/db/migration/V60__populate_bene_id_numeric.sql

```diff
@@ -0,0 +1,32 @@
+-- Script #2 of 3 flyway scripts to setup beneficiaries table with
+-- additional table column bene_id_numeric.
+
+-- HSQL always gets created/initialized from scratch so no need to drop trigger
+${logic.hsql-only} CREATE TRIGGER beneficiaries_insert_trigger BEFORE INSERT ON public.beneficiaries
+${logic.hsql-only}    REFERENCING NEW as newrow FOR EACH ROW
+${logic.hsql-only}       BEGIN ATOMIC
+${logic.hsql-only}            SET newrow.bene_id_numeric = convert(newrow.bene_id, SQL_BIGINT);
+${logic.hsql-only}       END;
+
+${logic.psql-only} CREATE OR REPLACE FUNCTION public.beneficiaries_pre_insert()
+${logic.psql-only}     RETURNS trigger
+${logic.psql-only}     LANGUAGE 'plpgsql'
+${logic.psql-only}    VOLATILE NOT LEAKPROOF
+${logic.psql-only} AS $BODY$
+${logic.psql-only} BEGIN
+${logic.psql-only}   NEW.bene_id_numeric = NEW.bene_id::bigint;
+${logic.psql-only}   RETURN NEW;
+${logic.psql-only} END;
+${logic.psql-only} $BODY$;
+
+${logic.psql-only} DROP TRIGGER IF EXISTS beneficiaries_insert_trigger ON public.beneficiaries;
+${logic.psql-only} CREATE TRIGGER beneficiaries_insert_trigger
+${logic.psql-only}     BEFORE INSERT ON public.beneficiaries
+${logic.psql-only}     FOR EACH ROW
+${logic.psql-only}       EXECUTE FUNCTION public.beneficiaries_pre_insert();
+
+-- update will be fairly long running; minimal effect readers (select)
+UPDATE public.beneficiaries
+${logic.hsql-only} SET bene_id_numeric = convert(bene_id, SQL_BIGINT)
+${logic.psql-only} SET bene_id_numeric = bene_id::bigint
+WHERE bene_id is not null;
```

Comment:
I thought in our last sync that we had planned to wrap this update in an exclusive lock as a best practice to protect against write conflicts.

Reply:
the lock will be held in this transaction which is doing the updates though

Reply:
OK, I was wrong here; you were correct that one can explicitly lock the table as part of the flyway transaction, but you still have to wrap the DDL within a transaction block. So I've made that change and tested.